docs/reference/contributing/target/lp_ticker.md

```diff
@@ -0,0 +1,70 @@
+
+<span class="warnings">**Warning:** We are changing the low power ticker HAL API in an upcoming release of Mbed OS. You can find details on how it may affect you in the [implementing the low power ticker API](#implementing-the-low-power-ticker-api) section.
+
+#### Assumptions
+
+##### Defined behavior
+- Has a reported frequency between 8KHz and 64KHz.
+- Has a counter that is at least 12 bits wide.
+- Continues operating in deep sleep mode.
+- The function `ticker_init` is safe to call repeatedly.
+- The function `ticker_init` allows the ticker to keep counting and disables the ticker interrupt.
+- Ticker frequency is nonzero and the counter is at least 8 bits.
+- The ticker rolls over at (1 << bits) and continues counting starting from 0.
+- The ticker counts at the specified frequency plus or minus 10%.
+- The ticker increments by 1 each tick.
+- The ticker interrupt fires only when the ticker times increments to or past the value that `ticker_set_interrupt` sets.
+- It is safe to call `ticker_set_interrupt` repeatedly before the handler is called.
+- The function `ticker_fire_interrupt` causes `ticker_irq_handler` to be called immediately from interrupt context.
+- The ticker operations `ticker_read`, `ticker_clear_interrupt`, `ticker_set_interrupt` and `ticker_fire_interrupt` take less than 20us to complete.
+
+##### Undefined behavior
+
+- Calling any function other than `ticker_init` before the initialization of the ticker.
+- Whether `ticker_irq_handler` is called a second time if the time wraps and matches the value that `ticker_set_interrupt again` sets.
+- Calling `ticker_set_interrupt` with a value that has more than the supported number of bits.
+- Calling any function other than `lp_ticker_init` after calling `lp_ticker_free`.
+
+##### Notes
+
+Be careful when encountering the following:
+
+- Drift due to reschedule.
+- Incorrect overflow handling of timers.
+- Interrupting at a time of 0.
+- Interrupt triggered more than once.
+
+#### Dependencies
+
+Hardware low power ticker capabilities.
+
+#### Implementing the low power ticker API
+
+We are working on the new HAL low power ticker API, which will replace current version in an upcoming release of Mbed OS. You need to implement the low power ticker API in both variants. First, you need to implement the current API. You can find it on master branch:
+
+[![View code](https://www.mbed.com/embed/?type=library)](https://os-doc-builder.test.mbed.com/docs/v5.7/mbed-os-api-doxy/lp__ticker__api_8h_source.html)
+
+To make sure your platform is ready for the upcoming changes, you need to implement the future API and submit it in a separate pull request against `feature-hal-spec-ticker` branch. You can find the API and specification for the new low power ticker API in the following header file:
+
+[![View code](https://www.mbed.com/embed/?type=library)](https://github.com/ARMmbed/mbed-os/blob/feature-hal-spec-ticker/hal/lp_ticker_api.h)
```

Comment:
Because we don't like linking to GH, I'll wait to merge this until we've reached a conclusion about feature branch Doxygen.